connectors/spark/src/main/scala/io/unitycatalog/spark/UCSingleCatalog.scala

```diff
@@ -180,6 +180,17 @@ class UCSingleCatalog
 object UCSingleCatalog {
   val LOAD_DELTA_CATALOG = ThreadLocal.withInitial[Boolean](() => true)
   val DELTA_CATALOG_LOADED = ThreadLocal.withInitial[Boolean](() => false)
+
+  def checkUnsupportedNestedNamespace(namespace: Array[String]): Unit = {
+    if (namespace.length > 1) {
+      throw new ApiException("Nested namespaces are not supported: " + namespace.mkString("."))
+    }
+  }
+
+  def fullTableNameForApi(catalogName: String, ident: Identifier): String = {
```

Comment:
please add docs explaining what this is doing with examples. what can be the various fields in ident, and what does it finally produce.